apps/ledger-live-mobile/src/components/ValidateError.tsx

```diff
@@ -1,19 +1,55 @@
-import React, { memo } from "react";
+import React, { memo, useCallback } from "react";
 import { View, StyleSheet } from "react-native";
-import { Trans } from "react-i18next";
-import { useTheme } from "@react-navigation/native";
+import { useTranslation } from "react-i18next";
+import { useTheme, useNavigation } from "@react-navigation/native";
 import GenericErrorView from "./GenericErrorView";
 import Button from "./Button";
 import NeedHelp from "./NeedHelp";
+import { BaseNavigation } from "./RootNavigator/types/helpers";
+import { NavigatorName, ScreenName } from "~/const";
+import { MANAGER_TABS } from "~/const/manager";
+import { LatestFirmwareVersionRequired } from "@ledgerhq/live-common/errors";
+import { RequiredFirmwareUpdate } from "./DeviceAction/rendering";
+import { useSelector } from "react-redux";
+import { lastConnectedDeviceSelector } from "~/reducers/settings";
+import { LedgerError } from "~/types/error";
 
 type Props = {
-  error: Error;
+  error: LedgerError;
   onClose: () => void;
   onRetry?: () => void;
 };
 
 function ValidateError({ error, onClose, onRetry }: Props) {
+  const navigation = useNavigation<BaseNavigation>();
+  const { t } = useTranslation();
   const { colors } = useTheme();
+
+  const managerAppName = error?.name === "UpdateYourApp" ? error.managerAppName : undefined;
```

Comment:
Should we do `error instanceof UpdateYourApp` instead of checking the name ?

Reply:
Yeah I was think of that too, I can change it to this if you want

Reply:
Updated